ethers-middleware/src/lib.rs

```diff
@@ -40,3 +40,18 @@ pub use timelag::TimeLag;
 /// [`Middleware`](ethers_providers::Middleware) in a concise way
 pub mod builder;
 pub use builder::MiddlewareBuilder;
+
+// For macro expansions only, not public API.
+
+#[doc(hidden)]
+#[allow(unused_extern_crates)]
+extern crate self as ethers;
+
+#[doc(hidden)]
+pub use ethers_contract as contract;
+
+#[doc(hidden)]
+pub use ethers_core as core;
+
+#[doc(hidden)]
+pub use ethers_providers as providers;
```

Comment:
hmm, I'm not sure I like that these are now always there.

Reply:
if you have a better way to find ethers crate names then go ahead, this is a patch for builds when that is not possible, and only for our internal crates

Reply:
I don't have a better solution for this, and seems like this is a proper fix. 
But let's document why we're doing this